src/azure-cli/az

```diff
@@ -16,4 +17,4 @@ else:
 if os.environ.get('AZ_INSTALLER') is None:
     os.environ['AZ_INSTALLER'] = 'PIP'
 
-os.execl(sys.executable, sys.executable, '-m', 'azure.cli', *sys.argv[1:])
+subprocess.run([sys.executable, '-m', 'azure.cli'] + sys.argv[1:])
```

Comment:
```suggestion
result = subprocess.run([sys.executable, '-m', 'azure.cli'] + sys.argv[1:])
sys.exit(result.returncode)
```
To fix the exit code of `az`. The previous code seems to always return `0` when running `echo $?` after a command run.

Reply:
This makes sense. However, I am not quite sure if the change is safe or robust enough. Maybe we can clearly say "calling `az` on Git Bash is not support"? Then let the user call `az.bat` (`C:\Users\xxx\AppData\Local\Programs\Python\Python38\Scripts\az.bat`) instead.